### ISSUE TITLE
Segfault when printing zmanim for northern cities

### ISSUE THREAD

**Issue by msalle** [hebcal/hebcal#263]
Calling `hebcal -Z` for a northern city will segfault for times in the summer, when alos haShachar cannot be calculated

How to reproduce:
Try for example `hebcal -Z -C Amsterdam 5 29 2023`, likewise for other northern cities.
which shows a stacktrace:
```
[...]
goroutine 1 [running]:
github.com/hebcal/hebcal-go/hebcal.formatTime(0x0?, 0x0?)
        hebcal-5.8.2/vendor/github.com/hebcal/hebcal-go/hebcal/candles.go:31 +0xe
github.com/hebcal/hebcal-go/hebcal.TimedEvent.Render({{{0x0, 0x0, 0x0, 0x0}, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0, ...}, ...}, ...)
        hebcal-5.8.2/vendor/github.com/hebcal/hebcal-go/hebcal/candles.go:86 +0x187
main.main()
        hebcal-5.8.2/main.go:538 +0x4ab
```

The problem is with the 16.1° alot haShachar that doesn't exist in summer time in those places. (note that in for example Amsterdam it's customary to use chatzos for those times).

Whether it should be solved here on in hebcal-go is a bit of a question.

Some background notes on how this gets about in the code:
- `AlotHaShachar()` in github.com/hebcal/hebcal-go/zmanim/zmanim.go calls `timeAtAngle()` in the same file.
- That one in turn calls `TimeOfElevation()` in github.com/nathan-osman/go-sunrise/elevation.go.
- The latter returns a zero Time object (`time.Time{}`) if it cannot calculate it. One can use for example .IsZero() to check, see https://pkg.go.dev/time#Time.IsZero

This gets converted into a zero TimedEvent object (`TimedEvent{}`):
- `dailyZemanim()` in github.com/hebcal/hebcal-go/hebcal/candles.go uses the return value of `AlotHaShachar()` in a call to `NewTimedEvent()` in the same file, that returns a zero TimedEvent object (`TimedEvent{}`) in case the time is zero.
- that event is added by `HebrewCalendar()` in github.com/hebcal/hebcal-go/hebcal/hebcal.go but from the returned `event.CalEvent` it's not really checkable any longer.

**Comment by mjradwin** [comment]
Thanks for finding this bug! We'll fix this so there isn't a crash.

**Comment by mjradwin** [comment]
Fixed in 5.8.3

```
$ ./hebcal --version
Hebcal version 5.8.3
$ ./hebcal -Z -C Amsterdam 5 29 2023
5/29/2023 9th of Sivan, 5783
5/29/2023 Misheyakir: 3:38
5/29/2023 Misheyakir Machmir: 3:55
5/29/2023 Dawn: 4:41
5/29/2023 Sunrise: 5:27
5/29/2023 Kriat Shema, sof zeman: 9:32
5/29/2023 Kriat Shema, sof zeman (MGA): 8:56
5/29/2023 Tefilah, sof zeman: 10:54
5/29/2023 Tefilah, sof zeman (MGA): 10:30
5/29/2023 Chatzot hayom: 1:37
5/29/2023 Mincha Gedolah: 2:18
5/29/2023 Mincha Ketanah: 6:23
5/29/2023 Plag HaMincha: 8:06
5/29/2023 Sunset: 9:48
5/29/2023 Dusk: 10:34
5/29/2023 Tzeit HaKochavim: 11:00
$
```
